plugin/nginx_plus_collectd.py

```diff
@@ -772,6 +834,42 @@ def get_processes(self):
         '''
         return self._send_get(self.processes_url)
 
+    def _get_api_version(self):
+        '''
+        Determines whether the Nginx-plus plugin has a versioned API or legacy API.
+        '''
+        newer_api_base_path = self.api_base_path if self.api_base_path is not None else '/api'
+        legacy_api_base_path = self.api_base_path if self.api_base_path is not None else '/status'
+        base_url = 'http://{}:{}'.format(self.status_host, str(self.status_port))
+
+        try:
+            response = requests.get("{}{}/{}".format(base_url, newer_api_base_path, DEFAULT_API_VERSION), auth=self.auth_tuple)
+            if response.status_code == requests.codes.ok:
+                return DEFAULT_API_VERSION
+            else:
+                response = requests.get("{}{}".format(base_url, legacy_api_base_path), auth=self.auth_tuple)
+                if response.status_code == requests.codes.ok:
+                    return 0
+            LOGGER.error(
+                "Failed to detect the Nginx-plus API type (versioned or legacy), please check your input configuration.")
+        except RequestException as e:
+            LOGGER.exception("Failed to detect the Nginx-plus API type (versioned or legacy), due to the error: %s", e)
+        return None
+
+    def validate_nginx_version(self):
+        '''
+        Detects the change in the Nginx version and raise an error in case of a version change or unable to get the version
+        '''
+        cur_nginx_version = self.get_nginx_version()
+
+        if cur_nginx_version is None:
+            LOGGER.error('Unable to get the Nginx version')
+            raise ValueError("Unable to get the Nginx version")
```

Comment:
Not seeing a reason to log and raise errors in the same function.  If the invoking function deems it necessary to log it should while treating the exceptional case.

Reply:
Makes sense. We have made the change accordingly.